### ISSUE TITLE
SES Illegal Address error

### ISSUE THREAD

**Issue by mxcl** [swift-aws/aws-sdk-swift#116]
Trying to send emails I get the error `Illegal address`, it appears that the final payload sent to AWS SES is wrong:

```
Action=SendEmail&Destination.ToAddresses.member.1=%5B%22mxcl@me.com%22%5D&Message.Body.Text.Data=…
```

Specifically `Destination.ToAddresses.member.1=%5B%22mxcl@me.com%22%5D` is the URL encoding for`[mxcl@me.com]` where the docs for SES do not enclose the string we provide in square brackets.

It should be: `Destination.ToAddresses.member.1=mxcl@me.com`

Using v2.0.0. Thanks.

Code:

```swift
let recipient = "foo@bar.com"

 _ = try ses.sendEmail(.init(
    destination: .init(toAddresses: [recipient]),
    source: "noreply@example.com",
    replyToAddresses: [],
    returnPath: "",
    message: .init(
    subject: .init(data: subject),
        body: .init(html: nil,
        text: .init(data: body))),
    sourceArn: "",
    returnPathArn: ""))
```

**Comment by mxcl** [comment]
Edit: I rolled back to `2.0.0-rc.6` for aws-sdk-swift-core.git which does not do this, but I still get the same error, so probably this is a false bug report, close if you like. Though it does seem wrong still that it is enclosing the string like that.

**Comment by mxcl** [comment]
K, magic, I made it this and it works

```
let recipient = "foo@bar.com"

 _ = try ses.sendEmail(.init(
    destination: .init(toAddresses: [recipient]),
    source: "noreply@example.com",
    message: .init(
    subject: .init(data: subject),
        body: .init(html: nil,
        text: .init(data: body)))))
```

But *not* with `2.0.0` only with the RC.

**Comment by jonnymacs** [comment]
@mxcl thanks for the report and for digging into it. 

Just to confirm - rolling back to rc-6 of the core lib, then removing `replyToAddresses`, `returnPath`, `sourceArn` and `returnPathArn` from the `SendEmailRequest` works?

**Comment by mxcl** [comment]
> Just to confirm - rolling back to rc-6 of the core lib, then removing replyToAddresses, returnPath, sourceArn and returnPathArn from the SendEmailRequest works?

Yes, the `illegal address` error stopped after doing these things and mail was successfully sent.

**Comment by adam-fowler** [comment]
I believe the changes in https://github.com/swift-aws/aws-sdk-swift-core/commit/bf7a4348ed85fe30e717810460d9f9ac911514cc fix this. This commit isn't in a release yet though.

**Comment by jonnymacs** [comment]
@adam-fowler from what I can tell bf7a434 is in the 2.0.0 tag of the core lib so I _think_ the report is that this is a bug introduced in the new `encodeToQueryDictionary`

@mxcl how did you install to get 2.0.0 of the core lib? did you edit your Package.swift locally?

**Comment by adam-fowler** [comment]
@jonnymacs Ok let me check

**Comment by adam-fowler** [comment]
Yep there is an issue. @mxcl have you ever had this working before?

**Comment by adam-fowler** [comment]
@jonnymacs I have a fix for this https://github.com/swift-aws/aws-sdk-swift-core/pull/56. Was a typo, involves a replacement of a variable.

**Comment by mxcl** [comment]
> @mxcl how did you install to get 2.0.0 of the core lib? did you edit your Package.swift locally?

My `Package.swift` was: 

     .package(url: "https://github.com/swift-aws/aws-sdk-swift.git", from: "2.0.2"),

Which gave this in `Package.resolved`:

```json
      {
        "package": "AWSSDKSwiftCore",
        "repositoryURL": "https://github.com/swift-aws/aws-sdk-swift-core.git",
        "state": {
          "branch": null,
          "revision": "aba1a0b4da354e8dbded0677b0b82571b4b8c32b",
          "version": "2.0.0"
        }
      },
```

To downgrade I edited `Package.resolved` and set it to RC6.

**Comment by mxcl** [comment]
> Yep there is an issue. @mxcl have you ever had this working before?

It worked until I did `swift package update`, and worked again once I reverted the update’s changes to `Package.resolved`.

**Comment by adam-fowler** [comment]
> It worked until I did `swift package update`, and worked again once I reverted the update’s changes to `Package.resolved`.

Interesting, output of arrays was completely wrong for queries before 2.0.0 rc6. Maybe it only worked for arrays with a single element. Anyway the issue in 2.0.0 is fixed. If you use AWSSDKSwiftCore 2.0.1 your issue should be resolved.

**Comment by mxcl** [comment]
Confirmed fixed in 2.0.1, many thanks!

**Comment by mxcl** [comment]
Also as an aside, it is truly heart warming that this library is so thorough and actively maintained, you people deserve much kudos. Thanks again!
